api/src/graphql/resolvers/index.js

```diff
@@ -38,8 +51,9 @@ const resolvers = {
     need(obj, { nodeId }, { driver }) {
       return findNodeByLabelAndId(driver, 'Need', nodeId);
     },
-    responsibilities(obj, { search }, { driver }) {
+    responsibilities(obj, { search, nodeId }, { driver }) {
       if (search) return searchRealities(driver, 'Responsibility', search);
+      if (nodeId) return findNodesByRelationshipAndLabel(driver, nodeId, 'FULFILLS', 'Responsibility', 'IN');
```

Comment:
I kept reading and saw how you use this further down in the UI code. Was there something that forced this change or was it just to simplify the UI code? If it was only to simplify I'm not sure I think the simplifications are worth making the GraphQL query less declarative, but if you disagree (given that you've spent time on the problem and have more info on it than I do), I'd say change the argument name. It should definitely not be `nodeId`. It could be something specific like `fulfillsNeedId` or something. Preferably a name that introduces a convention that we can reuse when/if adding arguments like this to other fields. What do you think?

Reply:
Yep, I ran into some issues with updating the cache for Responsibilities on the `REALITIES_UPDATE_SUBSCRIPTION`. When updating a responsibility nested inside a need object, the DetailView would go blank, its data object being null for some reason. I banged my head around it for a while until I tried to query straight for responsibilities instead of through a need object (as we don't need any data from the need object itself anyway), and it worked.

I seems to me that returning responsibilities without nesting them in a need object allows cache normalization in Apollo to function properly. And we just need to listen to the `REALITIES_UPDATE_SUBSCRIPTION` in the responsibilities container and update it there, and cache normalization will update the detail view also.

Good point on the argument name, I've changed it to `fulfillsNeedId`.